ebos/eclproblem.hh

```diff
@@ -2040,29 +2044,31 @@ public:
 
         // if requested, compensate systematic mass loss for cells which were "well
         // behaved" in the last time step
-        if (enableDriftCompensation_) {
+        // Note that we don't allow for drift compensation if there is
+        // no source terms i.e. no wells and no aquifers.
+        const auto& schedule = this->simulator().vanguard().schedule();
+        int episodeIdx = this->simulator().episodeIndex();
+        const auto& aquifer = this->simulator().vanguard().eclState().aquifer();
+        bool compensateDrift =  schedule.numWells(episodeIdx) > 0 && aquifer.active();
+        if (enableDriftCompensation_ && compensateDrift) {
             const auto& simulator = this->simulator();
             const auto& model = this->model();
 
             // we need a higher maxCompensation than the Newton tolerance because the
             // current time step might be shorter than the last one
-            Scalar maxCompensation = 10.0*model.newtonMethod().tolerance();
-
+            Scalar maxCompensation = model.newtonMethod().tolerance()/10;
```

Comment:
Maybe we should change the comment accordingly?

Reply:
done